core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -278,14 +286,26 @@ public void setConf(Object newConf) {
   @Override
   public List<TableIdentifier> listTables(SessionContext context, Namespace ns) {
     checkNamespaceIsValid(ns);
+    Map<String, String> queryParams = Maps.newHashMap();
+    List<TableIdentifier> tables = Lists.newArrayList();
+    String pageToken = "";
+    if (restPageSize != null) {
+      queryParams.put("pageToken", pageToken);
+      queryParams.put("pageSize", restPageSize);
```

Comment:
Shouldn't we only set `pageSize` if it's set?  This would result in an empty query param being sent and I don't think  we describe that behavior.  I would expect we omit the param since the server can also manage page size just through the opaque `pageToken`

Reply:
It looks like this was updated and is now set correctly.  Resolving.